src/core/p5.Renderer.js

```diff
@@ -116,6 +117,21 @@ class Renderer {
     this.currentShape.bezierVertex(position, textureCoordinates);
   }
 
+  splineEnds(mode) {
+    if (mode === undefined) {
+      return this.states.splineEnds;
+    } else {
+      this.states.splineEnds = mode;
+    }
+    this.updateShapeProperties();
+  }
+
+  splineVertex(x, y, z = 0, u = 0, v = 0) {
```

Comment:
Could we add logic here so that texture coordinates are only added if the renderer supports them?

Otherwise, the vertices will have a texture coordinate property even when the renderer created the current shape to omit texture coordinates from its `#vertexProperties`. Since it's the renderer's responsibility to decide what those properties are going to be at the outset, I'm thinking it should also be the renderer's responsibility to be consistent with its initial specification.

**Side note:**
Could we implement `vertexProperties` as a getter, instead of a regular method, or as `getVertexProperties()`? I feel like the name `vertexProperties()` is confusing because usually this kind of thing is a setter in p5. A benefit of declaring `getVertexProperties() {\* ... *\}` over `get vertexProperties() {\*...*\}` is that there might be a tendency to try to erroneously use the latter version as a setter at call sites. I know we're trying to move fast during this phase, but if you think a change would be helpful, I can file this under the clean-up task I already created. No need to refactor now.

Reply:
Update:
* Texture coordinates are now added only if the renderer supports them.
* Can rename to `getVertexProperties()` in another PR.